### PR TITLE
check to detect piping into anonymous functions

### PR DESCRIPTION
**Pull request by zachdaniel**
Adding this per the style guide here: https://github.com/lexmag/elixir-style-guide

### CODE REVIEW

**Comment by OvermindDL1** [comment]
I personally prefer piping into `case`'s.  ^.^;

```elixir
do_something
|> case do
  {:ok, value} -> blah(value)
  {:error, _reason} = err -> []
|> case do lst? -> more_things(List.wrap(lst)) end
```

Which I'm not sure is better than anon functions, but eh, less parenthesis...  >.>

**Comment by zachdaniel** [comment]
I've never really decided how I feel about piping into cases 🤷‍♂ . But this shouldn't stop that from happening anyway :)

**Comment by OvermindDL1** [comment]
I'm just thinking it might be good to have both in the same check... or perhaps different checks might be better...

**Comment by zachdaniel** [comment]
I think they are different enough to be separate checks. Case statements are a lot more common in pipelines as well, for good reason.

**Comment by whatyouhide** [comment]
Agreed, they feel like separate checks.

File: lib/credo/check/readability/pipes_to_anonymous_functions.ex [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      |> (fn i -> i * 2 end).()
+      |> (fn i -> i * 2 end).()
+  ... should be refactored to look like this:
+      defp timex_2(i), do: i * 2
+
+      foo
+      |> times_2()
+      |> times_2()
+  Piping into anonymous functions is cluttered and difficult to read.
+  Instead, define a private function and use that instead.
```
Comment (whatyouhide) [inline lib/credo/check/readability/pipes_to_anonymous_functions.ex]: Repetition of "instead" here. :)

```suggestion
  Instead, define a private function and use that.
```

**Comment by rrrene** [comment]
Hi, I merged this as a new default check for Credo, but somehow this PR does not reflect that (says something about "unknown repository"). Thx for your work on it!

**Pull request closed without merging**